### DISCUSSION TITLE
Setting ignore_errors to TRUE while using COPY to import CSV file with unterminated quotes

### DISCUSSION THREAD

**Discussion by parvizfarnia** [duckdb/duckdb#8820] [Q&A]
Hello everyone

I would like to ask a question about the `COPY` command.

**My environment:**
```
OS : Windows 11 (64 bits)
DuckDB version : 0.8.1 (64 bits)

```
As I understand, the parameter `quote` in the `COPY` command allows to escape characters. This is very important and useful, in particular, if among columns values in the given CSV file, there are values including the delimiter character (which in my case is pipe '|'). In my programs I use the default value for `quote`, that is, `quote '"'`.

This morning I received a huge CSV file that I had to import into a DuckDB database, however, the import via `COPY` failed as there were multiple rows (among millions) with unterminated quote. Fortunately, I didn't need those rows for my program, so I decided to simply ignore them by setting the `ignore_errors` parameter to `TRUE` in order to continue inserting only valid rows. However, even after adding this parameter, I had the very same error message and the import process stopped upon the first detected bad row in the file, as if I had not even provided the `ignore_errors` parameter in the call to the `COPY` function.

Here is a test case to reproduce the problem :

**DDL:**
```
drop table if exists tmptab;
create table tmptab(col_a varchar(5), col_b varchar(5));
```

**Data.txt :** 
This is the data file that we are going to import via `COPY`. At the 3rd line (skipping the header line) as you can see below, I added an unterminated quote in the 2nd column: 
```
col_a|col_b
a1|b1
a2|b2
a3|"
a4|b4
a5|b5

```

Now, let's import our file in the table we created above. Note that below we set `ignore_errors TRUE` to skip the 3rd row in our file which has an unterminated quote :

**DML:**
```
copy tmptab from 'C:/tmp/data.txt' (
    format csv, 
    delimiter '|',
    header TRUE, 
    NULL '', 
    quote '"', 
    encoding UTF8,
    ignore_errors TRUE
);

```

And the program stops with the following error message:

```
Error: Invalid Input Error: Error in file "C:/tmp/data.txt" on line 4: unterminated quotes. (  file=C:/tmp/data.txt
  delimiter='|'
  quote='"'
  escape='"' (default)
  header=1
  sample_size=20480
  ignore_errors=1
  all_varchar=0)

```

Is this a normal behaviour or it is me who doesn't know how to use the `IGNORE_ERRORS` parameter correctly?

Thanks in advance.

**Comment by Tishj** [comment]
`ignore_errors` doesn't work in all cases sadly

**Reply by parvizfarnia** [reply]
Currently, the only workaround that I have come up with, is to change the default character for the `quote` parameter. This morning, I set `quote '£'` instead of `quote '"'` and this solved the issue for now. But obviously, this is by no means a solid solution to the problem, what happens if the next time I have a CSV file including the '£' character? 

So, you confirm that it's not a bug in my program and it is a normal behaviour? Should I ask this as a new feature to add? I ask the question, because in fact there could be many situations in which the CSV file to be imported, is not clean and besides, it is not always easy (or even possible) to remove bad rows beforehand.
